midterm-2023-11-07/code/5041-filip-zhikovski/scripts.ts

```diff
@@ -0,0 +1,202 @@
+    const response = await fetch(url);
+    if (!response.ok) {
+
+        throw new Error("Response Not OK")
+    }
+
+    const data = await response.json();
+    return data;
+}
+
+
+
+const displayMovies = (movies: Movie[]) => {
+
+    const container = document.getElementById('movie-container')!;
+    container.innerHTML = "";
+    for (const movie of movies) {
+        const movieRow = generateMovieRow(movie);
+        container.appendChild(movieRow);
+    }
+}
+  
+    const generateMovieRow = (movie: Movie) => {
+        const row = document.createElement('div');
+        row.classList.add("movie-table")
+    
+        const idCell = document.createElement('div');
+        idCell.classList.add('movie-data', 'movie-id');
+        idCell.innerHTML = movie.id.toString();
+        row.appendChild(idCell);
+    
+        const titleCell = document.createElement('div');
+        titleCell.classList.add('movie-data', 'movie-title');
+        titleCell.innerHTML = movie.title;
+        row.appendChild(titleCell);
+
+        const directorCell = document.createElement('div');
+        directorCell.classList.add('movie-data', 'movie-director');
+        directorCell.innerHTML = movie.director;
+        row.appendChild(directorCell);
+       
+        const yearCell = document.createElement('div');
+        yearCell.classList.add('movie-data', 'movie-year');
+        yearCell.innerHTML = movie.year.toString();
+        row.appendChild(yearCell);
+        
+        const genreCell = document.createElement('div');
+        genreCell.classList.add('movie-data', 'movie-genre');
+        genreCell.innerHTML = movie.genre.toString();
+        row.appendChild(genreCell);
+
+        const plotCell = document.createElement('div');
+        plotCell.classList.add('movie-data', 'movie-plot');
+
+        if (movie.plot.charAt(50) === ' ') { 
+            const cutPlot = movie.plot.substring(0, 50);
+            plotCell.innerHTML = cutPlot + "...";
+        } else {
+            const index = movie.plot.indexOf(' ', 50);
+            if (index !== -1) {
+                const cutPlot = movie.plot.substring(0, index);
+                plotCell.innerHTML = cutPlot + "...";
+            } else {
+                const cutPlot = movie.plot.substring(0, 50);
+                plotCell.innerHTML = cutPlot + "...";
+            }
+        }
+        row.appendChild(plotCell);
+
+        const castCell = document.createElement('div');
+        castCell.classList.add('movie-data', 'movie-genre');
+        
+        const castInfo = movie.cast.map(actorInfo => `${actorInfo.actor} as ${actorInfo.character}`).join(', ');
+        castCell.innerHTML = castInfo;
+        row.appendChild(castCell);
+
+
+        const oscarsCell = document.createElement('div');
+        oscarsCell.classList.add('movie-data', 'movie-oscars');
+  
+        for (const category in movie.oscars) {
```

Comment:
Using `for..in` is ok, but I don't like the direct assignment of `oscarCell.innerHTML`, it's much cleaner if we calculate it separately, and then just set it.
+10